server/src/test/java/org/apache/druid/metadata/SqlSegmentsMetadataManagerTest.java

```diff
@@ -589,6 +590,156 @@ public void testMarkAsUsedNonOvershadowedSegments() throws Exception
     );
   }
 
+  @Test
+  public void testMarkAsUsedNonOvershadowedSegmentsInEternityIntervalWithVersions() throws Exception
```

Comment:
Should we also add tests to `IndexerSQLMetadataStorageCoordinatorTest`?

Reply:
Not really, as `IndexerSQLMetadataStorageCoordinator` only exposes `markSegmentsAsUnusedWithinInterval(String dataSource, Interval interval)`. Tests already exist for the non-versions variation.

The methods with versions reside in `SegmentsMetadataManager`, and `SqlSegmentsMetadataManagerTest` uses the concrete implementation, so we should get code coverage from all of these new tests.